### ISSUE TITLE
IPv6 static routes

### ISSUE THREAD

**Issue by elrido** [OpenRC/openrc#312]
I have an environment with a hand full of VMs with static IPv4 and IPv6 addresses. Until recently this has been a very straight forward setup with just a single gateway, so I just had to configure that one router as the default gateway on the network interfaces.

Now I have enabled IPv6 support on a docker VM, so I want to route a certain IPv6 subnet to that host. Because I use static IP configuration, I can't use radvd to advertise this new route for the other hosts to pick up. Hence I turned to the `/etc/conf.d/staticroute` file and using the `/etc/init.d/staticroute` script.

As far as I understand the logic in [staticroute](https://github.com/OpenRC/openrc/blob/54780a45826f789f1291742eb2c99be369bfe64c/init.d/staticroute.in#L91), if I populate the staticroute environment variable in `/etc/conf.d/staticroute` with "net 192.168.0.0 netmask 255.255.255.0 gw 10.73.1.1" (one of the provided example lines) the init script will actually run this command on start:
```bash
route add -net 192.168.0.0 netmask 255.255.255.0 gw 10.73.1.1
```
For my IPv6 case I therefore added "2001:ffff:ffff:1:4900::/79 gw 2001:ffff:ffff:1:4800::23" as the staticroute, which gets turned into:
```bash
route -A inet6 add -2001:ffff:ffff:1:4900::/79 gw 2001:ffff:ffff:1:4800::23
```
This fails to be interpreted by `route`, because of the dash that gets added.

My workaround on my systems is to change the [line 91 in /etc/init.d/staticroute](https://github.com/OpenRC/openrc/blob/54780a45826f789f1291742eb2c99be369bfe64c/init.d/staticroute.in#L91) to:
```bash
				route $family $xtra $2 $args
```
The implication is that I would now have to change any IPv4 routes to start with that dash that I removed from the script.

Given that this is probably a rare edge case and if changed, would break existing IPv4 routes configuration, I am not sure if this should be changed globally. Please let me know if you would accept a PR to document this in the commented examples in the /etc/conf.d/staticroute configuration file or feel free to close this issue. Alternatively I could provide a patch that prevents the dash to get added only for the IPv6 address family.

**Comment by user318** [comment]
Hi,

If I understand right, you shoud use "net" for IPv6 too, so you need for your example:
`net 2001:ffff:ffff:1:4900::/79 gw 2001:ffff:ffff:1:4800::23`
And your config is missing "net" in the beginning.

**Comment by user318** [comment]
OK, I see now the problem. But I am not a maintainer, I was just passing by. I could look how to deal with this issue and make a pull request, for example, to fix it.

**Comment by elrido** [comment]
Maybe I'll just provide a PR that implements a different logic in `/etc/init.d/staticroute` when `$family` is ipv6 and document this case in `/etc/conf.d/staticroute`. Then the maintainers can decide if it is worth a merge or to marginal of a case.

**Comment by axelu** [comment]
@elrido I don't think you have an edge case.  I have the very same issue.  But besides changing the logic in `/etc/init.d/staticroute`, there is in fact a solution. If you look at the code in `/etc/init.d/staticroute` you see it uses the `route` command unless you set `staticiproute` as the key, then it uses the `ip route` command. When it executes the `ip route` command, it does not prefix the args with the offending `-`. 

So simply add the following line to `/etc/conf.d/staticroute` instead:
`staticiproute="2001:ffff:ffff:1:4900::/79 gw 2001:ffff:ffff:1:4800::23"`  
Important is you use the keyword **`staticiproute`**, anything other than that will default to using `route` command.  It will then execute `ip route add 2001:ffff:ffff:1:4900::/79 gw 2001:ffff:ffff:1:4800::23` instead of using `route add` command.  

Otherwise I agree, the logic should be changed to not prefix the args with the `-` when it detected inet6 as the family, otherwise it should do it as it does today,  maybe that is a compromise to keep it backward compatible.

**Comment by prasanthnarasimha** [comment]
Hello, I've encountered this issue today and I can't get to the workaround discussed in this thread. 
I'm trying to add an ipv6 host to reject list in the Ip table with the following command
```
sudo route add -A inet6 -host fd00:ec2::254 reject

getaddrinfo: #host: -2
#host: Unknown host
```
Any suggestions for a workaround?

**Comment by elrido** [comment]
> Any suggestions for a workaround would be helpful!

The workaround was to drop `-net`, you use `-host`. Given the error message looks similar, you might hit the same incompatibility in net-tools' `route` regarding ipv6 address parsing? Let us know if it works sans `-host` parameter.

**Comment by prasanthnarasimha** [comment]
Yes dropping `-host` and `-net` helped work around this issue but there seems to be more ipv6 trouble with `route` (not related to this issue),
Thanks!.
```
[ec2-user~]$ sudo route -6 add fd00:ec2::254/128 metric 1 reject
SIOGIFINDEX: No such device

(works with ip route)
[ec2-user~]$ sudo ip -6 route add prohibit fd00:ec2::254 metric 1
[ec2-use ~]$ ip route show table all | grep fd00
prohibit fd00:ec2::254 dev lo metric 1 pref medium
```
